File: tests/performance_tests-py2many/mandelbrot/mandelbrot_2.jl
function mandelbrot(limit, c)::Int64
    z = 0 + 0im
    for i = 0:limit
        if abs(z) > 2
            return i
        end
        z = z * z + c
    end
    return i + 1
end

if abspath(PROGRAM_FILE) == @__FILE__
    @assert(mandelbrot(1, 0.2 + 0.3im) == 2)
    @assert(mandelbrot(5, 0.2 + 0.3im) == 6)
    @assert(mandelbrot(6, 0.2 + 0.3im) == 7)
    @assert(mandelbrot(10000, 1 + 0.3im) == 2)
    @assert(mandelbrot(10000, 0.6 + 0.4im) == 4)
end

File: tests/performance_tests/mandelbrot/mandelbrot_2.py
def mandelbrot(limit, c):
    z = 0 + 0j
    for i in range(limit + 1):
        if abs(z) > 2:
            return i
        z = z * z + c
    return i + 1


if __name__ == "__main__":
    assert mandelbrot(1, 0.2 + 0.3j) == 2
    assert mandelbrot(5, 0.2 + 0.3j) == 6
    assert mandelbrot(6, 0.2 + 0.3j) == 7
    assert mandelbrot(10000, 1 + 0.3j) == 2
    assert mandelbrot(10000, 0.6 + 0.4j) == 4
